File: Data Structures & Algorithms/is-anagram/submission-2.js
class Solution {
    /**
     * @param {string} s
     * @param {string} t
     * @return {boolean}
     */
    isAnagram(s, t) {
        if (s.length !== t.length) return false;

  const counts = new Map();

  for (const char of s) counts.set(char, (counts.get(char) ?? 0) + 1);
  for (const char of t) counts.set(char, (counts.get(char) ?? 0) - 1);

  for (const count of counts.values()) {
    if (count !== 0) return false;
  }

  return true;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counts = defaultdict(int)

        for num in nums:
            counts[num] += 1
        
        count_pairs = list(counts.items())
        count_pairs.sort(key=lambda item: -item[1])

        topk = [count_pairs[i][0] for i in range(k)]

        return topk

        